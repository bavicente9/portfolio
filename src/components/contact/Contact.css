
.contact{
    padding: 50px var(--GeneralMarginWidth);
    background-color: var(--DarkGray);
}

.contact>h1{
    text-align: center;
    margin:0;
    padding:0 0 25px 0;
    color: var(--DarkOrange);
    font-size: 2rem;
    text-transform: uppercase;
}
/*Personal message*/
.contact__personalMessage{
    padding: 15px 50px;
    margin:0 0 25px 0;
    display: flex;
    justify-content: left;
    align-items: center;
    gap: 50px;
    border: 2px solid var(--Orange);
    border-radius: 20px;
}

.contact__personalMessage p{
    font-family: 'Roboto', sans-serif;
    font-size: 1rem;
}

.contact__socialMedia {
    font-size: 1rem;
    display: flex;
    flex-direction: column;
    gap: 25px;
}

.contact__socialMediaLinks{
    display: flex;
    gap: 25px;
}

.contact__socialMedia a{
    width: 30px;
    height: 30px;
    cursor: pointer;
}

.contact__btnLinkedin{
    background:url(./icon-linkedin.svg) no-repeat ;
    background-size: contain;
}
.contact__btnGithub{
    background:url(./icon-github.svg) no-repeat ;
    background-size: contain;
}


/*form*/

.contact__form{
    border-radius: 25px;
    background: var(--GeneralBackGround);
    display: flex;
    flex-direction: column;
    gap: 25px;
    padding: 25px;
}

.contact__form label{
    display: flex;
    flex-direction: column;
    color: #fff;
    gap: 10px;
    text-transform: capitalize;
    font-size: 1rem;
    font-weight: 600;
}

.contact__form label input,.contact__form label textarea{
    padding:10px 10px ;
    border: none;
    border-radius: 5px;
    font-family: "Roboto", sans-serif;

}

.contact__form label textarea{
    resize: none;
    height: 200px;
}

.button-form{
    width: 200px;
}


@media (max-width:650px) {
    .contact{
        padding: 25px var(--GeneralRMarginWidth);
    }

    .contact__personalMessage{
        display: flex;
        flex-direction: column;
        align-items: start;
        gap: 0;
    }

    .button-form{
        place-self: center;
    }
 
}