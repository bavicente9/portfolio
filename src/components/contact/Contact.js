import './Contact.css'

export const Contact = ({infoJson}) => {
    //the form data is send to netlify.
    return (
        <section aria-label='contact' id='contact' className='contact'>
            <h1>Contact</h1>
            <div className='contact__container'>
                <div className='contact__personalMessage'>
                    <p>Feel free to send me a message, i,ll reply you as soon as possible.</p>
                    <div className='contact__socialMedia'>
                        <label><b>Email: </b>{infoJson.email}</label>
                      <div className='contact__socialMediaLinks'>
                        <a rel='external' href={infoJson.links.github} aria-label='github' className='contact__socialMediaBtn contact__btnGithub' ></a>
                        <a rel='external' href={infoJson.links.linkedin} aria-label='linkedin' className='contact__socialMediaBtn contact__btnLinkedin' ></a>
                      </div>
                    </div>
                </div>
                <form className='contact__form' name="contact" method="POST" data-netlify="true">
                    <label>Name:<input type="text" name="name" placeholder='name'    required /></label>
                    <label>email:<input type="email" name="email" placeholder='email' required/></label>
                    <label>message:<textarea name="message" placeholder='message'required /></label>
                    <button type="submit" className='button button-form'>Send</button>
                </form>
            </div>
        </section >
    )
}