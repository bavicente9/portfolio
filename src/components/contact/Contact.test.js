import { render } from '@testing-library/react';
import { Contact } from "./Contact"
describe('Skills', () => {
    const infoJson = JSON.parse( `{
        "email":"bavicente98@gmail.com",
        "links":{
            "github":"github.com",
            "linkedin":"linkedin.com"
        }   
    }`)
    it('renders correctly', () => {
        const { asFragment } = render(<Contact infoJson={infoJson}/>)
        expect(asFragment()).toMatchSnapshot()
    })
})