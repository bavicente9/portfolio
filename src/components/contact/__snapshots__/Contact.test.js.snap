// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders correctly 1`] = `
<DocumentFragment>
  <section
    aria-label="contact"
    class="contact"
    id="contact"
  >
    <h1>
      Contact
    </h1>
    <div
      class="contact__container"
    >
      <div
        class="contact__personalMessage"
      >
        <p>
          Feel free to send me a message, i'll reply you as soon as possible.
        </p>
        <div
          class="contact__socialMedia"
        >
          <label>
            <b>
              Email: 
            </b>
            bavicente98@gmail.com
          </label>
          <div
            class="contact__socialMediaLinks"
          >
            <a
              aria-label="github"
              class="contact__socialMediaBtn contact__btnGithub"
              href="github.com"
              rel="external"
            />
            <a
              aria-label="linkedin"
              class="contact__socialMediaBtn contact__btnLinkedin"
              href="linkedin.com"
              rel="external"
            />
          </div>
        </div>
      </div>
      <form
        class="contact__form"
        method="POST"
        name="contact"
        netlify="true"
      >
        <input
          name="form-name"
          type="hidden"
          value="contact"
        />
        <label>
          Name:
          <input
            name="name"
            placeholder="name"
            required=""
            type="text"
          />
        </label>
        <label>
          email:
          <input
            name="email"
            placeholder="email"
            required=""
            type="email"
          />
        </label>
        <label>
          message:
          <textarea
            name="message"
            placeholder="message"
            required=""
          />
        </label>
        <button
          class="button button-form"
          type="submit"
        >
          Send
        </button>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
